### PR TITLE
Add: fix celo as default chain before wallet is connected

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

On initial connection, always request to switch to Celo no matter the chain.
After connected, claim-button should work for all supported chains

About # (link your issue here)
https://github.com/GoodDollar/GoodProtocolUI/issues/316

### CODE REVIEW

File: packages/good-design/src/advanced/web3action/Web3Action.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ export const Web3ActionButton: FC<Web3ActionProps> = withTheme({
   // account/chainId changes and re-try to perform current step action
   useEffect(() => {
     const continueSteps = async () => {
-      if (!account) {
+      if (!account && !chainId) {
```
Comment (sirpy) [inline packages/good-design/src/advanced/web3action/Web3Action.tsx:102]: what happens if  there's no account connected? dont we need to ask user to connnect one?

File: packages/good-design/src/advanced/web3action/Web3Action.tsx [hunk 2]
```diff
@@ -108,9 +108,13 @@ export const Web3ActionButton: FC<Web3ActionProps> = withTheme({
       if (!supportedChains.includes(chainId ?? 0)) {
         setActionText(ButtonSteps.switch);
         await switchToChain(supportedChains[0]);
+        setActionText(ButtonSteps.action);
         return;
       }
 
+      // if there is already a switch request, don't do action
+      if (actionText === ButtonSteps.switch) return;
+
```
Comment (sirpy) [inline packages/good-design/src/advanced/web3action/Web3Action.tsx:117]: this code is needed only for this specific case? or is this a general fix?

**Review by sirpy — COMMENTED** [review]
Lets wait for the discussion wiith pat

File: packages/good-design/src/advanced/web3action/Web3Action.tsx (hunk 2 above)
Comment (johnsmith-gooddollar) [reply packages/good-design/src/advanced/web3action/Web3Action.tsx:117]: I'd suggest to set exact flag regarding "if there is already a switch request" state. for example use some ref

and of course, to check is already switching **before** switchToChain call

Comment (johnsmith-gooddollar) [inline packages/good-design/src/advanced/web3action/Web3Action.tsx:117]:
```suggestion
      // if there is already a switch request, don't do action
      if (switchingChainRef.current) {
        return;
      }  

      if (!supportedChains.includes(chainId ?? 0)) {
        setActionText(ButtonSteps.switch);
        switchingChainRef.current = true;
        await switchToChain(supportedChains[0]).finally(() => switchingChainRef.current = false);
        setActionText(ButtonSteps.action);
        return;
      }


```

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/good-design/src/advanced/web3action/Web3Action.tsx (hunk 2 above)
Comment (johnsmith-gooddollar) [reply packages/good-design/src/advanced/web3action/Web3Action.tsx:117]: (do not forget to add useRef)

**Pull request closed without merging**